corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -0,0 +1,117 @@
+                // API: https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md
+                displayControlsDefault: false,
+                boxSelect: true, // enables box selection
+                controls: {
+                    polygon: true,
+                    trash: true
+                },
+            });
+            map.addControl(draw);
+
+            map.on('draw.create', updateArea);
+            map.on('draw.delete', updateArea);
+            map.on('draw.update', updateArea);
+
+            function getCoordinates(event) {
+                return event.lngLat;
+            };
+
+            function updateArea(e) {
+                const data = draw.getAll();
+                const area = turf.area(data);
+                // Restrict the area to 2 decimal points.
+                const rounded_area = Math.round(area * 100) / 100;
+            }
+
+            // We should consider refactoring and splitting the below out to a new JS file
+            let clickedMarker;
+
+            self.addCaseMarkersToMap = function (cases) {
+                const markerColor = "#00FF00";
+                cases.forEach(element => {
+                    let coordinates = element.coordinates;
+                    if (coordinates && coordinates.lat && coordinates.lng) {
+                        self.addMarker(coordinates, markerColor);
+                    }
+                });
+            };
+
+            self.addMarker = function (coordinates, color) {
+                const marker = new mapboxgl.Marker({color: color, draggable: false});
+                marker.setLngLat(coordinates);
+                marker.addTo(map);
+            }
+
+            function moveMarkerToClickedCoordinate(coordinates) {
+                if (clickedMarker != null) {
+                    clickedMarker.remove();
+                }
+                if (draw.getMode() === 'draw_polygon') {
+                    // It's weird moving the marker around with the ploygon
+                    return;
+                }
+                clickedMarker = new mapboxgl.Marker({color: "FF0000", draggable: true});
+                clickedMarker.setLngLat(coordinates);
+                clickedMarker.addTo(map);
+            }
+
+            function showCreateCasePopup(coordinates) {
+                new mapboxgl.Popup()
+                .setLngLat(coordinates)
+                .setHTML(coordinates + '<br/><button> Create Case at this point </button>')
+                .addTo(map);
+            }
+
+            // Handle click events here
+            map.on('click', (event) => {
+                let coordinates = getCoordinates(event);
+                moveMarkerToClickedCoordinate(coordinates);
+            });
+            return self;
+        };
+
+        var map;
+
+        $(document).ajaxComplete(function () {
+            var mapDiv = $('#geospatial-map');
+            var $data = $(".map-data");
+
+            if (mapDiv.length && !map) {
```

Comment:
Adding the `!map` check, stuff works fine when loading the page intially. When applying the filters, the map disappears. Not sure why? When removing this check, I can see it tries to make multiple calls to mapbox. cc @orangejenny do you perhaps have wisdom for me here?

Reply:
Here is my understanding.

I think the reason that map is disappearing starting from second report load is that the map container is inside the reportcontent block which gets removed when the Apply button is clicked (I haven't verified this hypothesis). Subsequently the map doesn't loaded, because `map` variable already was defined in the first load, so not meeting `!map` condition.

For a solution, I am not 100% sure that below would work, but we could try one of the below

1. Unset the map variable completely whenever the user clicks 'Apply' button, you can do this by subscribing to click event of the 'Apply' button `$(document).on('click', '#apply-filters'...`. This should result in `!map` condition getting met on every report load.
2. Add the mapbox container outside of reportcontent block, in which case it shouldn't get cleared on subsequent loads. But in this case, you would have to reset the map by removing any markers on it that were previously there and any other data.

2 might be better if it works, since that way we don't have to re-render the Mapbox each time. To be consistent with the user experience of removing the table, we can hide the map on click event and show it back on ajaxComplete event with just the data updated.

Reply:
@sravfeyn Thanks for the suggestions.

> I think the reason that map is disappearing starting from second report load is that the map container is inside the reportcontent block which gets removed when the Apply button is clicked 

I couldn't find evidence for this, but it does seem to be the behaviour. I couldn't get 2 to work, and doing 1 (remounting) seemed to fix it, but in a non-ideal way: Multiple requests to mapbox are being made and sometimes the map still disappears, but not very often. I think this has to do with the order in which events fire. I updated the code for reference, but it will need to be "fixed". Any ideas?

Reply:
To close the loop here, we offlined and got 2 to work.

Reply:
[We added](https://github.com/dimagi/commcare-hq/pull/33020/commits/03a7a6692320d0de26d3f98cb0a7b188ee146985) a geospatial specific html base. I also updated the code to read coordinates from the `commcare_gps_point` property (format is "lat lng alt acc").

Here's a screenshot (1 case has a coordinate):
![image](https://github.com/dimagi/commcare-hq/assets/64970009/263594d5-c709-4798-b1f8-7cc1b40f776e)